spec/equality-with-mountebank.spec.js

```diff
@@ -276,6 +276,69 @@ testRunConfigs.forEach(config => {
       assert.equal(404, successResult.status, `Expected a failure response from [${fullMockUrl}], got a [${successResult.status}] for query string [${queryString}]`)
     }))
   })
+  it(`should be able to distinguish between the presence and absence of a query string (option A) (${config.name})`, async () => {
+    const url = '/example'
+    await setupImposters(config, {
+      port: mockPort,
+      protocol: 'http',
+      defaultResponse: { statusCode: 404, body: 'Default 404', headers: {} },
+      stubs: [
+        {
+          name: `With no query string`,
+          predicates: [
+            {
+              deepEquals: {
+                method: 'GET',
+                path: url,
+                query: {}
+              }
+            }
+          ],
+          responses: [
+            {
+              is: {
+                statusCode: 301
+              }
+            }
+          ]
+        },
+        {
+          name: `With a query string`,
+          predicates: [
+            {
+              deepEquals: {
+                method: 'GET',
+                path: url,
+                query: {
+                  a: 'b'
+                }
+              }
+            }
+          ],
+          responses: [
+            {
+              is: {
+                statusCode: 302
+              }
+            }
+          ]
+        }
+      ]
+    })
+
+    const fullMockUrl = config.mockedHttpBaseUrl + url
+    const matchesForQueryString = [
+      'a=b'
+    ]
+    const successResult = await fetch(fullMockUrl)
+
+    assert.equal(301, successResult.status, `Expected a success response from [${fullMockUrl}], got a [${successResult.status}] for no query string`)
+    await Promise.all(matchesForQueryString.map(async (queryString) => {
+      const successResult = await fetch(fullMockUrl + '?' + queryString)
+
+      assert.equal(302, successResult.status, `Expected a failure response from [${fullMockUrl}], got a [${successResult.status}] for query string [${queryString}]`)
```

Comment:
Why is this `Expected a failure response`? Shouldn't the fact that a 302 be regarded as a success response?

Reply:
You're right, I'll update that.